### DISCUSSION TITLE
go-mod-directive only works with 1.x versions but not 1.x.y versions

### DISCUSSION THREAD

**Discussion by chancez** [renovatebot/renovate#25475] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

docker.io/renovate/renovate:full

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

So I was hoping to remove our regexManager for managing the `go` directive in `go.mod` that we have, but ran into an issue where renovate reports `invalid-value` for the `currentValue` of `1.21.3` in our `go.mod`.

I'm running renovate like this:
```
docker run -ti -e LOG_LEVEL=debug -e GITHUB_COM_TOKEN="$(gh auth token)" -v /tmp:/tmp -v $(PWD):/usr/src/app docker.io/renovate/renovate:full renovate --platform=local | tee renovate.log
```

And our `go.mod`'s `go` directive looks like this:
```
module github.com/example/app

go 1.21.3
```

Here's a minimal reproducible example repo: https://github.com/chancez/renovate-go-mod-directive-repro

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: packageFiles with updates (repository=local)
       "config": {
         "gomod": [
           {
             "deps": [
               {
                 "managerData": {"lineNumber": 2},
                 "depName": "go",
                 "depType": "golang",
                 "currentValue": "1.21.3",
                 "datasource": "golang-version",
                 "versioning": "go-mod-directive",
                 "updates": [],
                 "packageName": "go",
                 "warnings": [],
                 "skipReason": "invalid-value",
                 "fixedVersion": "1.21.3"
               }
             ],
             "packageFile": "go.mod"
           }
         ]
       }
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment, marked as answer]
Thanks for the report and reproduction. Fix is in #25479

**Reply by itchyny** [reply]
How can we stop it widen the Go version? I mean Renovate updates `go 1.20` into `go 1.21.0`, but I expect `go 1.21`.

**Reply by rarkins** [reply]
@itchyny please start a new discussion question. Do not raise it as a bug report unless you are certain that Renovate itself is updating the go directive or if it's `go` doing it
